### ISSUE TITLE
Failing to download artifact -> 500 error

### ISSUE THREAD

**Issue by AndyTempel** [go-gitea/gitea#25630]
### Description

Using the github.com/actions/download-artifact@v3 I'm unable to get the artifact from the previous job. In the gitea logs I found the following errors:
```
2023/07/02 14:26:22 ...eb/routing/logger.go:102:func1() [I] router: completed GET /api/actions_pipeline/_apis/pipelines/workflows/6/artifacts/5/download for my.ip.address:0, 500 Internal Server Error in 133.6ms @ <autogenerated>:1(actions.artifactRoutes.downloadArtifact-fm)
2023/07/02 14:13:58 ...ons/runner/runner.go:148:UpdateTask() [W] Failed to find the sent outputs of task 12: pq: relation "action_task_output" does not exist
```

in the gitea act I found this:

```
[build-docker-image/build-image]   | ::debug::Artifact Url: http://xx.xx.xx.xx:9081/api/actions_pipeline/_apis/pipelines/workflows/6/artifacts?api-version=6.0-preview
...
[build-docker-image/build-image]   | A 500 response code has been received while attempting to download an artifact
[build-docker-image/build-image]   | Exponential backoff for retry #1. Waiting for 5506 milliseconds before continuing the download
...
```

After multiple attempts it fails.

### Gitea Version

1.20.0+rc2-25-g1ddfe0313

### Can you reproduce the bug on the Gitea demo site?

No

### Git Version

2.40.1, Wire Protocol Version 2 Enabled

### Operating System

Docker, Unraid 6.12.2

### How are you running Gitea?

Using the 1.20-nightly docker image. 
Gitea Act: v0.2.1+1-gf2629f2
For storage, I use minio.
Database is postgres-14.

### Database

PostgreSQL

**Comment by AndyTempel** [comment]
Possibly fixed by: https://github.com/go-gitea/gitea/pull/24874
Workaround:

1. Before uploading tar the directory/files
2. Upload artifact
3. Download artifact and untar

**Comment by lunny** [comment]
> Possibly fixed by: #24874 Workaround:
> 
>     1. Before uploading tar the directory/files
> 
>     2. Upload artifact
> 
>     3. Download artifact and untar

Could you help to confirm #24874 fix that?

**Comment by lunny** [comment]
Closed by #24874 merged.
